diff --git a/build.sbt b/build.sbt
index 7870073..7124bd3 100644
--- a/build.sbt
+++ b/build.sbt
@@ -6,7 +6,11 @@ name := "scala-arm"
 
 version := "1.1"
 
-scalaVersion := "2.9.1"
+scalaVersion := "2.10.0-virtualized-SNAPSHOT"
+
+scalaHome := Some(file("/home/namin/code/scala-dev/build/pack/")) // TODO: continuations plugin doesn't work without this
+
+resolvers += MavenRepository("Local Maven Repo", (Path.userHome / ".m2" / "repository").asURL.toString)
 
 autoCompilerPlugins := true
 
